### PR TITLE
typecheckin_with_proptypes translated

### PR DESCRIPTION
**Pull request by lastsaintcrucified**
Please check my translation of typechecking with proptypes

### CODE REVIEW

**Comment by shakib609** [comment]
Please follow [this](https://github.com/reactjs/bn.reactjs.org/wiki/Glossary) while translating.

File: content/docs/typechecking-with-proptypes.md [hunk 1]
```diff
@@ -6,13 +6,13 @@ redirect_from:
   - "docs/react-api.html#typechecking-with-proptypes"
 ---
 
-> Note:
+> বিঃদ্রঃ:
 >
-> `React.PropTypes` has moved into a different package since React v15.5. Please use [the `prop-types` library instead](https://www.npmjs.com/package/prop-types).
+> `React.PropTypes` - অন্য একটি প্যাকেজে সরিয়ে নেয়া হয়েছে React v15.5 এর পর থেকে। দয়া করে এটি ব্যাবহার করুন [ `prop-types`লাইব্রেরী ব্যাবহার করুন](https://www.npmjs.com/package/prop-types).
```
Comment (shakib609) [inline content/docs/typechecking-with-proptypes.md]:
```suggestion
> `React.PropTypes` - অন্য একটি প্যাকেজে সরিয়ে নেয়া হয়েছে React v15.5 এর পর থেকে। দয়া করে [`prop-types` লাইব্রেরীটি ব্যবহার করুন](https://www.npmjs.com/package/prop-types)।
```

File: content/docs/typechecking-with-proptypes.md [hunk 2]
```diff
@@ -6,13 +6,13 @@ redirect_from:
...
 >
->We provide [a codemod script](/blog/2017/04/07/react-v15.5.0.html#migrating-from-reactproptypes) to automate the conversion.
+>আমরা এটি  [একটি কোডমড স্ক্রিপ্ট](/blog/2017/04/07/react-v15.5.0.html#migrating-from-reactproptypes) দিয়ে স্বয়ংক্রিয় কনভারসেশন চালাই।
```
Comment (shakib609) [inline content/docs/typechecking-with-proptypes.md]:
```suggestion
>আমরা [একটি codemod স্ক্রিপ্ট](/blog/2017/04/07/react-v15.5.0.html#migrating-from-reactproptypes) সরবরাহ করি যা দিয়ে স্বয়ংক্রিয়ভাবে কনভারশন চালাই।
```

File: content/docs/typechecking-with-proptypes.md [hunk 3]
```diff
@@ -6,13 +6,13 @@ redirect_from:
...
 
-As your app grows, you can catch a lot of bugs with typechecking. For some applications, you can use JavaScript extensions like [Flow](https://flow.org/) or [TypeScript](https://www.typescriptlang.org/) to typecheck your whole application. But even if you don't use those, React has some built-in typechecking abilities. To run typechecking on the props for a component, you can assign the special `propTypes` property:
+আপনার অ্যাপ যত বড় হতে থাকবে টাইপচেকিং এ তত ভুল ধরা পড়তে থাকবে। কিছু অ্যাপ এর ক্ষেত্রে, টাইপচেকের জন্য আপনি এরকম [ফ্লো](https://flow.org/) or [টাইপস্ক্রিপ্ট](https://www.typescriptlang.org/) কিছু জাভাস্ক্রিপ্ট এক্সটেনশন ব্যাবহার করতে পারেন। আপনি যদি এগুলো ব্যাবহার করতে না চান তাহলে React এর কিছু বিল্ট-ইন টাইপচেকিং এবিলিটি আছে যা আপনি ব্যাবহার করতে পারেন। একটি কম্পোনেন্ট এর জন্য props এর উপর টাইপচেকিং করতে চাইলে আপনি স্পেশাল `propTypes` প্রোপার্টি এসাইন করতে পারেন:
```
Comment (shakib609) [inline content/docs/typechecking-with-proptypes.md]:
```suggestion
আপনার অ্যাপ যত বড় হতে থাকবে টাইপচেকিং এ তত ভুল ধরা পড়তে থাকবে। কিছু অ্যাপের ক্ষেত্রে, টাইপচেকের জন্য আপনি এরকম [Flow](https://flow.org/) অথবা [TypeScript](https://www.typescriptlang.org/) এর মত কিছু জাভাস্ক্রিপ্ট এক্সটেনশন ব্যবহার করতে পারেন। আপনি যদি এগুলো ব্যবহার করতে না চান তাহলে React এর কিছু বিল্ট-ইন টাইপচেকিং এবিলিটি আছে যা আপনি ব্যাবহার করতে পারেন। একটি কম্পোনেন্টের জন্য props এর উপর টাইপচেকিং করতে চাইলে আপনি স্পেশাল `propTypes` প্রোপার্টি এসাইন করতে পারেনঃ
```

File: content/docs/typechecking-with-proptypes.md [hunk 4]
```diff
@@ -30,11 +30,12 @@ Greeting.propTypes = {
 };
 ```
 
-`PropTypes` exports a range of validators that can be used to make sure the data you receive is valid. In this example, we're using `PropTypes.string`. When an invalid value is provided for a prop, a warning will be shown in the JavaScript console. For performance reasons, `propTypes` is only checked in development mode.
+`PropTypes` অনেকগুলো ভ্যালিডেটরস এক্সপোর্ট করে যা দিয়ে যাচাই করা হয় আপনার ডাটাগুলো ভ্যালিড কিনা। এই উদাহারণে আমরা `PropTypes.string` ব্যাবহার করছি। যখন কোন ইনভ্যালিড ভ্যালু prop এর জন্য দেয়া হয় তখন জাভাস্ক্রিপ্ট কন্সোল এ একটি সতর্কবার্তা প্রদর্শন করা হয়। পারফরম্যান্স এর স্বার্থে শুধুমাত্র ডেভেলপম্যান্ট মোড এ `propTypes` চেক করা হয়।
```
Comment (shakib609) [inline content/docs/typechecking-with-proptypes.md]:
```suggestion
`PropTypes` অনেকগুলো ভ্যালিডেটরস এক্সপোর্ট করে যা দিয়ে যাচাই করা হয় আপনার ডাটাগুলো ভ্যালিড কিনা। এই উদাহরণে আমরা `PropTypes.string` ব্যবহার করছি। যখন কোন ইনভ্যালিড ভ্যালু prop এর জন্য দেয়া হয় তখন জাভাস্ক্রিপ্ট কন্সোলে একটি সতর্কবার্তা প্রদর্শন করা হয়। পারফরম্যান্সের স্বার্থে শুধুমাত্র ডেভেলপমেন্ট মোডে `propTypes` চেক করা হয়।
```

File: content/docs/typechecking-with-proptypes.md [hunk 5]
```diff
@@ -30,11 +30,12 @@ Greeting.propTypes = {
...
 
-### PropTypes {#proptypes}
+### প্রপটাইপ্স {#proptypes}
```
Comment (shakib609) [inline content/docs/typechecking-with-proptypes.md]: No Need to translate this. This is a Keyword in React

File: content/docs/typechecking-with-proptypes.md [hunk 6]
```diff
@@ -30,11 +30,12 @@ Greeting.propTypes = {
...
 
-Here is an example documenting the different validators provided:
+
```
Comment (shakib609) [inline content/docs/typechecking-with-proptypes.md]: Extra newline

File: content/docs/typechecking-with-proptypes.md [hunk 7]
```diff
@@ -30,11 +30,12 @@ Greeting.propTypes = {
...
+এখানে প্রদত্ত বিভিন্ন ভ্যালিডেটরের ডকুমেন্টিং এর একটি উদাহরণ রয়েছে:
```
Comment (shakib609) [inline content/docs/typechecking-with-proptypes.md]: : => ঃ

File: content/docs/typechecking-with-proptypes.md [hunk 8]
```diff
@@ -128,9 +129,8 @@ MyComponent.propTypes = {
 };
 ```
 
-### Requiring Single Child {#requiring-single-child}
-
-With `PropTypes.element` you can specify that only a single child can be passed to a component as children.
+### রিক্যুয়েরিং সিংগেল চাইল্ড {#requiring-single-child}
```
Comment (shakib609) [inline content/docs/typechecking-with-proptypes.md]: Not sure about this. What do you suggest, @nutboltu vai?

File: content/docs/typechecking-with-proptypes.md [hunk 9]
```diff
@@ -128,9 +128,8 @@ MyComponent.propTypes = {
...
+`PropTypes.element` এর মাধ্যমে আপনি একটি সিংগেল চাইল্ড একটি নির্দিষ্ট কম্পোনেন্ট এ পাস করতে পারবেন।
```
Comment (shakib609) [inline content/docs/typechecking-with-proptypes.md]: Missing newline

File: content/docs/typechecking-with-proptypes.md [hunk 10]
```diff
@@ -6,13 +6,13 @@ redirect_from:
   - "docs/react-api.html#typechecking-with-proptypes"
 ---
 
-> Note:
+> নোট:
 >
-> `React.PropTypes` has moved into a different package since React v15.5. Please use [the `prop-types` library instead](https://www.npmjs.com/package/prop-types).
+> `React.PropTypes` - অন্য একটি প্যাকেজে সরিয়ে নেয়া হয়েছে React v15.5 এর পর থেকে। দয়া করে এটি ব্যাবহার করুন [the `prop-types` library instead](https://www.npmjs.com/package/prop-types).
```
[the remark this answers is not in the document]
Comment (lastsaintcrucified) [reply content/docs/typechecking-with-proptypes.md]: Thanks for the suggestion.

File: content/docs/typechecking-with-proptypes.md [hunk 11]
```diff
@@ -6,13 +6,13 @@ redirect_from:
...
 >
->We provide [a codemod script](/blog/2017/04/07/react-v15.5.0.html#migrating-from-reactproptypes) to automate the conversion.
+>আমরা এটি  [a codemod script](/blog/2017/04/07/react-v15.5.0.html#migrating-from-reactproptypes) দিয়ে স্বয়ংক্রিয় কনভারসেশন চালাই।
```
[the remark this answers is not in the document]
Comment (lastsaintcrucified) [reply content/docs/typechecking-with-proptypes.md]: Thanks brother

File: content/docs/typechecking-with-proptypes.md (hunk 1 above)
Comment (lastsaintcrucified) [reply content/docs/typechecking-with-proptypes.md]: Thanks brother

File: content/docs/typechecking-with-proptypes.md (hunk 3 above)
Comment (lastsaintcrucified) [reply content/docs/typechecking-with-proptypes.md]: Thanks brother

File: content/docs/typechecking-with-proptypes.md (hunk 4 above)
Comment (lastsaintcrucified) [reply content/docs/typechecking-with-proptypes.md]: Thanks brother

File: content/docs/typechecking-with-proptypes.md (hunk 7 above)
Comment (lastsaintcrucified) [reply content/docs/typechecking-with-proptypes.md]: Thanks brother

File: content/docs/typechecking-with-proptypes.md (hunk 8 above)
Comment (lastsaintcrucified) [reply content/docs/typechecking-with-proptypes.md]: Thanks brother

File: content/docs/typechecking-with-proptypes.md (hunk 6 above)
Comment (lastsaintcrucified) [reply content/docs/typechecking-with-proptypes.md]: Thanks brother

**Pull request merged**